client/state/comments/actions.js

```diff
@@ -182,35 +179,18 @@ export const changeCommentStatus = ( siteId, postId, commentId, status ) => ( {
 	status,
 } );
 
-export function editComment( siteId, postId, commentId, content ) {
-	return dispatch => {
-		dispatch( {
-			type: COMMENTS_EDIT,
-			siteId,
-			postId,
-			content,
-		} );
-
-		return wpcom
-			.site( siteId )
-			.comment( commentId )
-			.update( { content } )
-			.then( data =>
-				dispatch( {
-					type: COMMENTS_EDIT_SUCCESS,
-					siteId,
-					postId,
-					commentId,
-					content: data.content,
-				} )
-			)
-			.catch( () =>
-				dispatch( {
-					type: COMMENTS_EDIT_FAILURE,
-					siteId,
-					postId,
-					commentId,
-				} )
-			);
-	};
-}
+/**
+ * Creates an action that edits a comment.
+ * @param {Number} siteId Site identifier
+ * @param {Number} postId Post identifier
+ * @param {Number} commentId Comment identifier
+ * @param {Object} comment New comment data
+ * @returns {Object} Action that edits a comment
+ */
+export const editComment = ( siteId, postId, commentId, comment ) => ( {
```

Comment:
Maybe we can also reference `https://developer.wordpress.com/docs/api/1.1/post/sites/%24site/comments/%24comment_ID/` and the shape of the expected comment fields.

Reply:
Added a reference to the API docs on the data-layer instead, since the action creator should not concern itself with neither comment shape nor side effects... I think 🤔

Reply:
Maybe not the api ref, but the adding the comment shape in js doc would be helpful for callers

Reply:
I've added two `@typedef` for _Author_ and _Comment_ with the properties we allow (or expect to) edit operations on.